src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -885,7 +913,7 @@ internal static bool IsValidExtensionDataProperty(JsonPropertyInfo jsonPropertyI
                 // Avoid a reference to typeof(JsonNode) to support trimming.
                 (memberType.FullName == JsonObjectTypeName && ReferenceEquals(memberType.Assembly, typeof(JsonTypeInfo).Assembly));
 
-            return typeIsValid && jsonPropertyInfo.Options.GetConverterFromTypeInfo(memberType) != null;
```

Comment:
Second component no longer relevant -- `DetermineConverter()` will resolve a converter for the property instance at a later stage.